advanced/sr_gui_fingertip_visualization/src/sr_gui_fingertip_visualization/visual_tab_layouts.py

```diff
@@ -0,0 +1,439 @@
+                                                   1.95, -1.95, 0.00]
+        self._coordinates['v2']['sensing']['y'] = [4.38, 6.38, 10.78, 11.50, 14.50, 15.08, 16.00, 17.00, 19.00, 21.00,
+                                                   4.38, 6.38, 10.78, 11.50, 14.50, 15.08, 16.00, 17.00, 19.00, 21.00,
+                                                   7.38, 9.50, 9.50, 11.20]
+        self._coordinates['v2']['excitation'] = dict()
+        self._coordinates['v2']['excitation']['x'] = [5.30, 6.00, -5.30, -6.00]
+        self._coordinates['v2']['excitation']['y'] = [9.00, 22.00, 9.00, 22.00]
+        self._electrode_count = len(self._coordinates[self._version]['sensing']['x'])
+
+    def _tactile_data_callback(self, data):
+        for data_field in self._data_fields:
+            for i, finger in enumerate(self._fingers):
+                if data_field == "pac0":
+                    self._data[finger][data_field] = data.tactiles[i].pac0
+                elif data_field == "pac1":
+                    self._data[finger][data_field] = data.tactiles[i].pac1
+                elif data_field == "pac":
+                    self._data[finger][data_field] = list(data.tactiles[i].pac)
+                elif data_field == "pdc":
+                    self._data[finger][data_field] = data.tactiles[i].pdc
+                elif data_field == "tac":
+                    self._data[finger][data_field] = data.tactiles[i].tac
+                elif data_field == "tdc":
+                    self._data[finger][data_field] = data.tactiles[i].tdc
+                elif data_field == "electrodes":
+                    self._data[finger][data_field] = list(data.tactiles[i].electrodes)
+
+    def _init_tactile_layout(self):
+        self.finger_complete_layout = QHBoxLayout()
+        for finger in ['th', 'ff', 'mf', 'rf', 'lf']:
+            self.finger_complete_layout.addWidget(self._init_finger_widget(finger))
+        self.setLayout(self.finger_complete_layout)
+
+    def _init_finger_widget(self, finger):
+        self._finger_frame[finger] = QGroupBox(finger)
+        self._finger_frame[finger].setCheckable(True)
+        self._finger_frame[finger].setSizePolicy(1, 1)
+        layout = QVBoxLayout()
+
+        x_cords = self._coordinates[self._version]['sensing']['x']
+        y_cords = self._coordinates[self._version]['sensing']['y']
+        min_x = abs(min(self._coordinates[self._version]['sensing']['x']))
+        min_y = abs(min(self._coordinates[self._version]['sensing']['y']))
+        max_x = abs(max(self._coordinates[self._version]['sensing']['x']))
+        max_y = abs(max(self._coordinates[self._version]['sensing']['y']))
+        y_offset = 2
+
+        detected_fingertip_type = self._detect_biotac_type(finger)
+
+        if detected_fingertip_type == BiotacType.SP_PLUS:
+            self._finger_widgets[finger] = [QWidget()] * len(self._coordinates[self._version]['sensing']['x'])
```

Comment:
Why initializing the lsit with empty QWidgets if it's to assign Dotunit a few lines below?